permit time overlaps

autotel opened the pull request:
Permit time overlaps (chronicity is checked comparing start points and not end/start)
also help a bit more in the compiler error

autotel committed:
Update compiler.js

Fix checking for time order.
* permit overlap of cues (not comparing end to start, but start to start)

autotel committed:
Try to fix lint related issues

autotel commented:
forgot that space after the comma, but you can get to that :)

osk commented:
Hi @autotel, thanks for the PR! Can you fix the lint issue so the tests run?

Also I'm adding a failing test case for each bug to make sure we don't regress. Can you add that as well?

autotel committed:
Added test for overlapping cues allowance.
Allow overlapping cues, which is done by comparing starting points instead of end-versus start points.
Perhaps a warning should be emitted in case of overlapping cues?
